src/compiler/checker.ts

```diff
@@ -21742,6 +21743,10 @@ namespace ts {
             return type.flags & TypeFlags.Union ? every((<UnionType>type).types, f) : f(type);

```

Comment:
Is there a reason why this doesn't iterate over constituents of intersection types too?